display_list/skia/dl_sk_paint_dispatcher.h

```diff
@@ -37,7 +37,19 @@ class DlSkPaintDispatchHelper : public virtual DlOpReceiver {
   void setMaskFilter(const DlMaskFilter* filter) override;
   void setImageFilter(const DlImageFilter* filter) override;
 
-  const SkPaint& paint() { return paint_; }
+  const SkPaint& paint() {
+    // On the Impeller backend, we will only support dithering of *gradients*,
+    // and it will be enabled by default (without the option to disable it).
+    // Until Skia support is completely removed, we only want to respect the
+    // dither flag for gradients (otherwise it will also apply to, for
+    // example, images, which is not supported in Impeller).
+    //
+    // See https://github.com/flutter/flutter/issues/112498.
+    if (color_source_gradient_ && dither_) {
```

Comment:
I don't think you want this inside an if, as it will get turned on once and never turned off. Just call `paint_.setDither(this && that)` in all cases. paint_ is reused across multiple rendering ops.

Reply:
Done! And thanks for catching this, I didn't realize it was re-used like that!